semantic_segmentation/scripts/train.py

```diff
@@ -1,23 +1,78 @@
-import os
+# import os
 
-os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
-os.environ["CUDA_VISIBLE_DEVICES"] = "1"
+# os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
+# os.environ["CUDA_VISIBLE_DEVICES"] = "1"
 
 import argparse
-from datetime import datetime
 
 import matplotlib.pyplot as plt
 import numpy as np
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks.early_stopping import EarlyStopping
+from pytorch_lightning.loggers.tensorboard import TensorBoardLogger
 import torch
 import torch.nn as nn
 import torch.optim as optim
-from augment import get_transform
-from dataset import Dataset
-from torch.utils.tensorboard import SummaryWriter
 from torchmetrics import Accuracy
-from tqdm import tqdm
 from unet_bn import U2NET_lite
 
+from augment import get_transform
+from dataset import Dataset
+
+
+class UNET_LT(pl.LightningModule):
+    def __init__(self, lr, **kwargs):
+        super().__init__()
+        self.Net = U2NET_lite()
+        self.Net.cuda(1)
+        self.loss = nn.CrossEntropyLoss(ignore_index=255)
+        self.lr = lr
+        self.accuracy = Accuracy(
+            task="multiclass", num_classes=16, ignore_index=255
+        ).cuda(1)
+        self.save_hyperparameters()
+
+    @staticmethod
+    def add_model_specific_args(parent_parser):
+        parser = parent_parser.add_argument_group("U2NetModel")
+        parser.add_argument("--lr", type=float, default=3e-4)
+        return parent_parser
+
+    def forward(self, x):
+        pred = self.Net(x)
+        return pred
+
+    def configure_optimizers(self):
+        optimizer = optim.RAdam(self.parameters(), lr=self.lr)
+        return optimizer
+
+    def training_step(self, train_batch, batch_idx):
+        img, target = train_batch
+        img = img.cuda(1)
```

Comment:
Remove explicit .cuda
Lighting automatically copies the data - your batch should be already on gpu

If you want to use gpu1 instead of first check Trainer arg - there you can set how many gpus or just pass the gpu ids